### ISSUE TITLE
Add option to include sound in the exported video

### ISSUE THREAD

**Issue by hamoid** [hamoid/video_export_processing#11]

**Comment by Jacowaco** [comment]
Hi :) is there an easy way to add sound?

**Comment by hamoid** [comment]
There's a quick fix to add sound, just by changing the ffmpeg command arguments that are being called. But doing it well may require more effort. For instance: 

* What if the sound is shorter than the video? Should the sound loop? 
* What if the sound is longer than the video? Should it just be cut at video-length?
* Should the program automatically stop when the video length equals sound length?

I have a more advanced idea, but then it changes the goal of this library: it could provide an fft analysis of the sound file, so then this library would become a tool to produce sound-driven video clips. But maybe it's better to keep it separate from the library. I could just add an example to achieve this.

**Comment by Jacowaco** [comment]
Great answer!
That kind of programming exceeds my Processing skills, but i imagine that when people thinks in video, they think in both image and sound, so i guess it wouldn't really change the goal of the library if you add some sound features.

**Comment by softpunch** [comment]
Could you add an option to remove the "-an" parameter in ffmpeg? The library doesn't have to include *every* audio option to be useful as a video+audio recorder... especially considering there isn't really a simple alternative provided elsewhere.

**Comment by hamoid** [comment]
Hi @softpunch , I think what I could do is to add a method to replace "-an" by anything you want. That way you could add any sound file to your video file. I'll do that when I get some free time.

**Comment by softpunch** [comment]
@hamoid Smart! That sure beats having to route audio through SoundFlower to Quicktime for a screen cap. (Remember when Jack worked on OSX? That was nice...)

**Comment by clankill3r** [comment]
The advanced sound idea sounds better for a different library.
I think the default behaviour should be:
-video makes up the length, if audio is longer the audio get's cut off.
-if audio is shorter then the last part won't have audio

**Comment by hamoid** [comment]
The feature is now available on the last version. I did not research how hard it would be to add sound while exporting each frame, so what I did was to attach the sound file to the video file at the end, after exporting all frames. This means calling ffmpeg twice. 

Unfortunately I re-encode the audio, because padding with silence was not possible without re-encoding (it failed to run). I prefer lossless copy, but that did not work if the audio was shorter than the video. If I did not do audio padding, the video player time slider acted funny: it stopped moving at the end of the audio, even if video kept playing. Pull requests with better approaches are welcome :)
